tests/test_approval_gate.py

```
@@ -0,0 +1,232 @@
+        return self.outcome
+
+
+class _FakeTask:
+    def __init__(self):
+        self.frames: List[Any] = []
+
+    async def queue_frame(self, frame):
+        self.frames.append(frame)
+
+
+async def _execute_recorder(record: List[str], result: Any = {"ok": True}):
```

Ruff (0.15.15)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)